test-network-function/networking/suite.go

```diff
@@ -275,3 +293,96 @@ func testNodePort(env *config.TestEnvironment) {
 		}
 	})
 }
+
+func getDeclaredPortList(container int, podName, podNamespace string) {
+	protocolName, port, name := "", "", ""
+	ocCommandToExecute := fmt.Sprintf(commandportdeclared, podName, podNamespace, container)
+	res, _ := utils.ExecuteCommand(ocCommandToExecute, ocCommandTimeOut, interactive.GetContext(false))
+	x := strings.Split(res, "\n")
+	for _, i := range x {
+		if strings.Contains(i, "containerPort") {
+			s := strings.Split(i, ":")
+			s = strings.Split(s[1], ",")
+			port = s[0]
+		}
+		if strings.Contains(i, "name") {
+			s := strings.Split(i, ":")
+			s = strings.Split(s[1], ",")
+			name = s[0]
+		}
+		if strings.Contains(i, "protocol") {
+			s := strings.Split(i, ":")
+			protocolName = s[1]
+		}
+		if port == "" || name == "" || protocolName == "" {
+			continue
+		}
+		p, _ := strconv.Atoi(strings.TrimSpace(port))
+		name = strings.TrimSpace(strings.ReplaceAll(name, "\"", ""))
+		protocolName = strings.TrimSpace(strings.ReplaceAll(protocolName, "\"", ""))
+		declaredPort[p] = name + " " + protocolName
+		port, name, protocolName = "", "", ""
+	}
+}
+
+func getListeningPortList(commandlisten []string, nodeOc *interactive.Context) {
```

Comment:
Same thing with this function.  I think it should return the actual value instead of setting the global variable.

Reply:
I fixed.